dragonfly/item/hoe.go

```diff
@@ -0,0 +1,82 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/internal/item_internal"
+	"github.com/df-mc/dragonfly/dragonfly/item/tool"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// Hoe is a tool generally used to till dirt and grass blocks into FarmLand blocks for planting crops.
+// Additionally a Hoe can be used to break certain types of blocks such as Crimson and Hay Blocks.
+type Hoe struct {
+	Tier tool.Tier
+}
+
+// This function will turn a dirt or grass block into a farmland if the necessary properties are met.
+func (h Hoe) UseOnBlock(pos world.BlockPos, face world.Face, clickPos mgl64.Vec3, w *world.World, user User, ctx *UseContext) bool {
+	if grass := w.Block(pos); grass == item_internal.Grass || grass == item_internal.Dirt {
+		if face == world.FaceDown {
+			// Tilled land isn't created when the bottom face is clicked.
+			return false
+		}
+		if w.Block(pos.Add(world.BlockPos{0, 1})) != item_internal.Air {
+			// Tilled land can only be created if air is above the grass block.
+			return false
+		}
+		w.SetBlock(pos, item_internal.FarmLand)
+		//TODO: Play the sound
```

Comment:
Don't forget to implement playing the sound. Let us know in Discord if you need help with that.